#83 modify esnI18nService.translate method

tuanlc opened the pull request:
Consistent the input of esnI18nService.translate and angular translate filter, the second param (if present) is an object that contains pairs of key + value

itswillta commented:
Can we have 2 commits with one of them just to remove the IIFEs?

itswillta commented on src/frontend/js/modules/i18n/i18n-string.service.js:
```diff
@@ -1,25 +1,29 @@
+        if (this.ignoreSanitizeStrategy) {
+          this.translated = $translate.instant(this.text, this.interpolateParams, undefined, undefined, null);
+        } else {
+          this.translated = $translate.instant(this.text, this.interpolateParams);
```
```suggestion
      if (this.translated) return this.translated;

      this.translated = this.ignoreSanitizeStrategy ? $translate.instant(this.text, this.interpolateParams, undefined, undefined, null) : $translate.instant(this.text, this.interpolateParams);
      return this.translated;
```

tuanlc replied:
Your solution is lack of step to assign the translated value to this.translated

piavgh commented:
> Can we have 2 commits with one of them just to remove the IIFEs?

+1 for this. It's hard to see what are the core changes.

itswillta commented on src/frontend/js/modules/i18n/i18n.service.js:
```diff
@@ -1,64 +1,67 @@
+    function getFullLocale(callback) {
+      return esnConfig('core.language', ESN_I18N_DEFAULT_LOCALE)
+      .then(function(locale) {
+        var fullLocale = ESN_I18N_FULL_LOCALE.hasOwnProperty(locale) ? ESN_I18N_FULL_LOCALE[locale] : ESN_I18N_DEFAULT_FULL_LOCALE;
```
This is old code, but can you change this to `const`.

itswillta commented on src/frontend/js/modules/i18n/i18n.service.js:
```diff
@@ -1,64 +1,67 @@
+      getFullLocale: getFullLocale,
+      translate: translate,
+      isI18nString: isI18nString
+    };
```
```suggestion
    return {
      getLocale,
      getFullLocale,
      translate,
      isI18nString
    };
```

itswillta commented on src/frontend/js/modules/i18n/i18n.service.js:
```diff
@@ -1,64 +1,67 @@
-          var params = (arguments.length > 1) ? Array.prototype.slice.call(arguments).slice(1) : [];
+    function getFullLocale(callback) {
+      return esnConfig('core.language', ESN_I18N_DEFAULT_LOCALE)
+      .then(function(locale) {
```
This is old code but wrong indentation.

Commit:
#82 remove IIFE

Commit:
#83 modify esnI18nService.translate method

Consistent the input of esnI18nService.translate and angular translate filter, the second param (if present) is an object that contains pairs of key + value

Commit:
#83 ES6ify the i18n.service.js file

tuanlc commented:
I separated commits

piavgh commented on src/frontend/js/modules/i18n/i18n.service.js:
```diff
@@ -25,15 +25,20 @@ angular.module('esn.i18n')
+     */
+    // TODO: Write tests for this (https://github.com/OpenPaaS-Suite/esn-frontend-common-libs/issues/83)
+    function translate(text, interpolateParams = {}, ignoreSanitizeStrategy = false) {
       if (!text || text instanceof EsnI18nString) {
```
If text = null or undefined, it should return an empty string, right?

tuanlc replied:
Not sure since this is an old logic. Maybe this is intended at the beginning.
Bc, there is no bug for this so far. I will keep it like this.

Commit:
#83 remove i18n interpolator service since it is not used anymore
